pkg/asset/installconfig/alibabacloud/alibabacloud.go

```diff
@@ -41,12 +41,19 @@ func selectRegion(client *Client) (string, error) {
 	}
 	regions := regionsResponse.Regions.Region
 
+	regionMap := map[string]string{}
 	longRegions := make([]string, 0, len(regions))
 	shortRegions := make([]string, 0, len(regions))
 	for _, location := range regions {
 		longRegions = append(longRegions, fmt.Sprintf("%s (%s)", location.RegionId, location.LocalName))
 		shortRegions = append(shortRegions, location.RegionId)
+		regionMap[location.RegionId] = location.LocalName
 	}
+	_, ok := regionMap[defaultRegion]
+	if !ok {
+		return "", errors.Errorf("installer bug: invalid default alibabacloud region %q", defaultRegion)
+	}
```

Comment:
I don't think there is much value in using a map here.
```suggestion
	var defaultLongRegion string
	longRegions := make([]string, 0, len(regions))
	shortRegions := make([]string, 0, len(regions))
	for _, location := range regions {
		longRegion := fmt.Sprintf("%s (%s)", location.RegionId, location.LocalName)
		longRegions = append(longRegions, longRegion)
		shortRegions = append(shortRegions, location.RegionId)
		if location.RegionId == defaultRegion {
			defaultLongRegion = longRegion
		}
	}
```

Reply:
Have  changed^^